frontend/lib/util/hook/useDebouncedCallback.ts

```diff
@@ -0,0 +1,22 @@
+import { useCallback, useEffect, useRef } from "react";
+
+export function useDebouncedCallback<T extends (...args: never[]) => unknown>(
+  callback: T,
+  delay: number,
+) {
+  const debounceTimerRef = useRef(0);
+  useEffect(() => {
+    return () => {
+      window.clearTimeout(debounceTimerRef.current);
+    };
+  }, []);
+
+  return useCallback(
+    (...args: Parameters<T>) => {
+      window.clearTimeout(debounceTimerRef.current);
+      debounceTimerRef.current = window.setTimeout(() => callback(...args), delay);
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [delay],
```

Comment:
Can we somehow not disable this lint?

Reply:
You mean globally?

That's not recommended: https://github.com/facebook/create-react-app/issues/6880#issuecomment-485912528

Reply:
No, I want to keep the lint active instead of disabling it.

Reply:
It complains about `callback` not being a dependency. If we add `callback` as a dependency, it stops working. I think it was because the callback then gets recreated every time, causing the `debounceTimerRef` to reset (not working right now, so I can't check to be sure).

Reply:
Looking at @jschuurk-kr's link I wouldn't be too sure that, even though skipping `callback` in the deps works, it is also safe. Let's look into how to improve this!

Reply:
This is something we ran into all the time with react, you cannot add all dependencies, because if there is one that changes, but the effect should not run every time, you cannot add the changing dep to dependencies because it will cause a never-ending loop.

Reply:
Wrapping the callback in a ref (https://github.com/kiesraad/abacus/pull/146/files#r1688191696) seems to work, not sure if this has any other implications though.